src/hooks/services/common/useAuthenticationState.ts

```diff
@@ -0,0 +1,35 @@
+import { useRouter } from 'next/router';
+import { useSession } from 'next-auth/react';
+import { useEffect, useState } from 'react';
+import type { AuthenticationState } from 'types/authentication';
+import { PAGE_PATH } from 'constants/common';
+
+export const useAuthenticationState = () => {
```

Comment:
p3;
State 라는 단어 제거해도 좋을 것 같아요!
```suggestion
export const useAuthentication = () => {
...
  const [authentication, setAuthentication] =
    useState<Authentication>({
      user: undefined,
      status: 'loading',
    });
 ...
}
```

Reply:
p2;

이미 비슷한 동작을 하는 `useUser`(`src/hooks/common/useUser.ts`)라는 훅이 존재하고 있습니다!
아래는 `useUser` 훅 코드 입니다.

```js
export const useUser = (): LoggedStatus => {
  const router = useRouter();
  const { data: session, status } = useSession();

  const isLoading = status === 'loading';
  const isAuthenticated = status === 'authenticated';
  const isUnauthenticated = status === 'authenticated';

  const useData = session !== null ? session?.user : null;

  if (isUnauthenticated) {
    void router.push(PAGE_PATH.account.login);
  }

  return {
    user: useData,
    isLoading,
    isLoggedIn: isAuthenticated,
  };
};

```

그런데 찾아보니 `useUser`를 사용하는 부분이 없는 것 같네요.
(@Bori-github 보리님이 생성하신 것 같은데 어떤 목적으로 만들었는지 확인해주시면 좋을 것 같아요.)


우상님이 필요하신 데이터를  `useUser`에 추가하거나 만드신 `useAuthenticationState` 을 유지하거나 하나로 통일됐으면 좋겠습니다!

Reply:
@sohyeonAn @yws1502 
아 useUser는 제가 따로 테스트 해보려고 생성했다가 실수로 함께 포함된 내용이네요ㅜㅜ
useUser는 제거하겠습니다!

Reply:
이미 비슷한 훅이 있는지 몰랐네요... ㅎㅎ
@sohyeonAn  @Bori-github useUser 훅은 제가 useAuthentication 훅 수정하면서 삭제하겠습니다!

작업 커밋 : 7ce22f5

Reply:
@sohyeonAn @yws1502 
제가 이 전에 작업했던 다른 브랜치에서 항상 non-null session 값을 반환하도록 적용해본다고 했는데, 시도해보니 제가 생각했던 거랑 다르고 의도한대로 동작하지 않아서 현재로는 우상님이 생성해주신 커스텀 훅 사용하는 것이 더 적합한 것 같아요!

틈틈히 더 좋은 방법 고민해볼게요!